Packet++/header/BgpLayer.h

```diff
@@ -0,0 +1,114 @@
+#ifndef PACKETPP_BGP_LAYER
+#define PACKETPP_BGP_LAYER
+
+#include "Layer.h"
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+	static const size_t BGP_MARKER_SIZE_BYTES = 16;
+
+	/**
+	 * @struct bgp_header
+	 * Represents a BGP protocol header
+	 */
+#pragma pack(push, 1)
+	struct bgp_header {
+		/** Reserved marker */
+		uint8_t marker[BGP_MARKER_SIZE_BYTES];
+
+		/** Length (in bytes) of the message, including this header */
+		uint16_t messageLength;
+
+		/** Type of message */
+		uint8_t messageType;
+	};
+#pragma pack(pop)
+
+	/**
+	* An enum of all supported BGP message types
+	*/
+	enum BgpMessageType
+	{
+		BGP_UNKNOWN,
+
+		BGP_OPEN,
+		BGP_UPDATE,
+		BGP_NOTIFICATION,
+		BGP_KEEP_ALIVE,
+		BGP_ROUTE_REFRESH,
+
+		BGP_NUM_MESSAGE_TYPES
+	};
+
+	/**
+	 * @class BgpLayer
+	 * Represents a BGP layer
```

Comment:
It's worth mentioning which version of the BGP protocol is being supported here. Probably version 4?